### PR TITLE
Add uuid serch with pymisp

### PR DESCRIPTION
**Pull request by truckydev**
the uuid need to validate this regex : ('^[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}\Z', re.I)

### CODE REVIEW

File: pymisp/api.py [hunk 1]
```diff
@@ -368,6 +368,19 @@ def change_threat_level(self, event, threat_level_id):
...
+
+            CakeText::uuid follow RFC 4122
+             - the third group must start with a 4,
+             - the fourth group must start with 8, 9, a or b.
+             
+             :param uuid: A UUID to validate
+        """
+        regex = re.compile('^[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}\Z', re.I)
+        match = regex.match(uuid)
+        return bool(match)
```
Comment (FloatingGhost) [inline pymisp/api.py]: You don't need to do this - you're evaluating this in an if statement anyhow - you can just return regex.match

File: pymisp/api.py [hunk 2]
```diff
@@ -368,6 +368,19 @@ def change_threat_level(self, event, threat_level_id):
 
         return self._check_response(response)
 
+    def _valid_uuid(self,uuid):
+        """ 
+            Test valid uuid générate by CakePhp
```
Comment (FloatingGhost) [inline pymisp/api.py]: Any chance you can translate these before merge?

File: pymisp/api.py [hunk 3]
```diff
@@ -746,6 +759,7 @@ def search(self, values=None, not_values=None, type_attribute=None,
             :param date_from: First date
             :param date_to: Last date
             :param last: Last updated events (for example 5d or 12h or 30m)
+            :param uuid: A uuid valideted
```
Comment (FloatingGhost) [inline pymisp/api.py]: Same here, spell check / translate

File: pymisp/api.py (hunk 2 above)
Comment (truckydev) [inline pymisp/api.py]: yes maybe something like : 

Test if uuid is valid

Comment (FloatingGhost) [inline pymisp/api.py]: No it's fine - the comment... it's just french :D

File: pymisp/api.py (hunk 1 above)
Comment (truckydev) [inline pymisp/api.py]: Your right,

I refere to this stackoverflow for this job : 
http://stackoverflow.com/questions/11384589/what-is-the-correct-regex-for-matching-values-generated-by-uuid-uuid4-hex

Your solution seems more elegant  ^^

File: pymisp/api.py (hunk 3 above)
Comment (truckydev) [inline pymisp/api.py]: ok

**Comment by truckydev** [comment]
If you merge, I let you close the issue : 
https://github.com/MISP/MISP/issues/1683

Thanks

**Comment by FloatingGhost** [comment]
Had to do a bit of an interesting merge - there were a few conflicts because you didn't pull before you pushed

**Comment by FloatingGhost** [comment]
but added-  thank

**Pull request merged**